src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -612,15 +607,67 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
                 await LoopbackServer.CreateServerAsync(async (origServer, origUrl) =>
                 {
                     var request = new HttpRequestMessage(new HttpMethod(oldMethod), origUrl);
-                    if (oldMethod == "POST")
+
+                    Task<HttpResponseMessage> getResponseTask = client.SendAsync(request);
+
+                    await LoopbackServer.CreateServerAsync(async (redirServer, redirUrl) =>
                     {
-                        request.Content = new StringContent(ExpectedContent);
+                        // Original URL will redirect to a different URL
+                        Task<List<string>> serverTask = origServer.AcceptConnectionSendResponseAndCloseAsync((HttpStatusCode)statusCode, $"Location: {redirUrl}\r\n");
+
+                        await Task.WhenAny(getResponseTask, serverTask);
+                        Assert.False(getResponseTask.IsCompleted, $"{getResponseTask.Status}: {getResponseTask.Exception}");
+                        await serverTask;
+
+                        // Redirected URL answers with success
+                        serverTask = redirServer.AcceptConnectionSendResponseAndCloseAsync();
+                        await TestHelper.WhenAllCompletedOrAnyFailed(getResponseTask, serverTask);
+
+                        List<string> receivedRequest = await serverTask;
+
+                        string[] statusLineParts = receivedRequest[0].Split(' ');
 
-                        if (useTE)
+                        using (HttpResponseMessage response = await getResponseTask)
                         {
-                            request.Headers.TransferEncodingChunked = true;
+                            Assert.Equal(200, (int)response.StatusCode);
+                            Assert.Equal(newMethod, statusLineParts[0]);
                         }
-                    }
+                    });
+                });
+            }
+        }
+
+        [Theory]
+        [InlineData(300)]
+        [InlineData(301)]
+        [InlineData(302)]
+        [InlineData(303)]
+        public async Task AllowAutoRedirect_True_PostToGetDoesNotSendTE(int statusCode)
+        {
+            if (IsCurlHandler)
+            {
+                // ISSUE #27301:
+                // CurlHandler incorrectly sends Transfer-Encoding when the method changes from POST to GET.
+                // Also, note CurlHandler doesn't change POST to GET for 300 response, either (see above test)
+                return;
+            }
+
+            if (IsWinHttpHandler)
+            {
+                // This test occasionally fails on WinHttpHandler.
```

Comment:
Can you add the issue number here.

Reply:
I meant to but forgot.  Thanks for catching.